src/StoreBase.ts

```diff
@@ -60,178 +56,174 @@ export abstract class StoreBase {
-        if (this._bypassThrottle) {
-            let pendingThrottledStore = this._pendingThrottledStores.shift();
-            while (!!pendingThrottledStore) {
-                pendingThrottledStore._resolveThrottledCallbacks();
-                pendingThrottledStore = this._pendingThrottledStores.shift();
-            }
-        }
+        
+        StoreBase._resolveCallbacks();
     }
 
-    constructor(throttleMs: number = 0, bypassTriggerBans = false) {
-        this._throttleMs = throttleMs;
-        this._bypassTriggerBlocks = bypassTriggerBans;
+    constructor(private readonly _throttleMs?: number, private readonly _bypassTriggerBlocks = false) {
     }
 
     // If you trigger a specific set of keys, then it will only trigger that specific set of callbacks (and subscriptions marked
     // as "All" keyed).  If the key is all, it will trigger all callbacks.
     protected trigger(keyOrKeys?: string|number|(string|number)[]) {
-        let keys: string[]|undefined;
+        let throttleMs: number | undefined;
+        if (this._throttleMs !== undefined) {
+            throttleMs = this._throttleMs;
+        } else {
+            // If the store doens't define any throttling, pick up the default
+            throttleMs = Options.defaultThrottleMs;
+        }
 
-        // trigger(0) is valid, ensure that we catch this case
-        if (keyOrKeys || _.isNumber(keyOrKeys)) {
-            keys = _.map(_.isArray(keyOrKeys) ? keyOrKeys : [keyOrKeys], key => _.isNumber(key) ? key.toString() : key);
+        // If we're throttling, save execution time
+        let throttledUntil: number | undefined;
+        if (throttleMs) {
+            if (!this._throttleData) {
+                // Needs to accumulate and trigger later -- start a timer if we don't have one running already
+                // If there are no callbacks, don't bother setting up the timer
+                this._throttleData = {
+                    timerId: Options.setTimeout(this._handleThrottledCallbacks, this._throttleMs),
+                    callbackTime: Date.now() + throttleMs
+                };
+            }
+            throttledUntil = this._throttleData.callbackTime;
         }
 
-        // Build a list of callbacks to call, trying to accumulate keys into a single callback set to avoid multiple callbacks
-        // to the same target with different keys.
+        const bypassBlock = this._bypassTriggerBlocks;
 
-        if (!keys) {
+        // trigger(0) is valid, ensure that we catch this case
+        if (!keyOrKeys && !_.isNumber(keyOrKeys)) {
             // Inspecific key, so generic callback call
             const allSubs = _.flatten(_.values(this._subscriptions));
-            _.forEach(allSubs, callback => {
-                // Clear the key list to null for the callback
-                this._gatheredCallbacks.set(callback, null);
-            });
+            const handleCallback = (callback: SubscriptionCallbackFunction) => {
```

Comment:
Move this to a separate private function to keep from constantly closure-ing it?  Doesn't appear to really rely on any context from the outer function that it's modifying.